Assets/Shopify/UIToolkit/Theme Controllers/CartController.cs

```diff
@@ -0,0 +1,122 @@
+namespace Shopify.UIToolkit {
+    using UnityEngine.Events;
+    using Shopify.Unity;
+    using Shopify.Unity.SDK;
+    using System.Linq;
+
+    public class CartController {
+        [System.Serializable]
+        public class QuantityChangeEvent : UnityEvent<int> {}
+
+        [System.Serializable]
+        public class PurchaseFailedEvent : UnityEvent<ShopifyError> {}
+
+        private Cart Cart;
+        public UnityEvent OnPurchaseStarted = new UnityEvent();
+        public QuantityChangeEvent OnQuantityChange = new QuantityChangeEvent();
+        public UnityEvent OnPurchaseComplete = new UnityEvent();
+        public UnityEvent OnPurhcaseCancelled = new UnityEvent();
+        public PurchaseFailedEvent OnPurhcaseFailed = new PurchaseFailedEvent();
+
+        public CartController(Cart cart) {
+            SetCart(cart);
+        }
+
+        public void SetCart(Cart cart) {
+            Cart = cart;
+        }
+
+        public CartLineItems LineItems {
+            get {
+                return Cart.LineItems;
```

Comment:
Should be noted that currently if you go and modify quantities directly on line items we're not reactive to this. Will need to create a second PR which adds eventing to the `Cart` that propagates line item changes.